### ISSUE TITLE
Bug: TypeError: Cannot set property 'good' of undefined

### ISSUE THREAD

**Issue by amilajack** [xjamundx/eslint-plugin-promise#34]
I'm using eslint `3.8.1`. When running eslint, i get the following error:

```
Cannot set property 'good' of undefined
TypeError: Cannot set property 'good' of undefined
    at EventEmitter.markCurrentBranchAsGood (/Users/amila/Documents/Projects/parallel.js/node_modules/eslint-plugin-promise/rules/always-return.js:75:52)
```

**Comment by r-murphy** [comment]
I find that this happens with unreachable code. Specifically when there is a throw or a return on unreachable code.

```js
function foo() {
  return bar().then(x => {
    return x;
    return "unreachable"; //this one causes the error
  });
}
```

I have atom-eslint enabled and configured to lint as I type. So when I'm changing the logic and temporarily put in a return above some other code I get a bunch of error popups from atom.

**Comment by xjamundx** [comment]
Your fix looks good, can you file a PR?

**Comment by xjamundx** [comment]
let me publish this out

**Comment by joeybaker** [comment]
I'm getting the same error (with v3.3.1). Is it possible this didn't get published? A fresh install doesn't seem to include the fix in #39

**Comment by xjamundx** [comment]
Let me check

**Comment by xjamundx** [comment]
Published `3.3.2` but I think the code should have been there with `3.3.1` we may need to reopen the issue

**Comment by joeybaker** [comment]
For what it's worth, when I do a fresh install of v3.3.2, I see the fix from #39. Thanks!
